### PR TITLE
GAV tag should be mandatory in CloudSlang java_action

### PR DESCRIPTION
**Pull request by genadi-hp**
GAV tag should be mandatory in CloudSlang java_action

Sign-off: @genadi-hp, Genadi Rabinovich, genadi@hpe.com

### CODE REVIEW

**Comment by orius123** [comment]
user must work with GAV also if he uses the CLI?

**Comment by genadi-hp** [comment]
Yes, user must work with GAV also if he uses the CLI - user should provide information where (in which jar) to search for the specified java class. We want to have control on it.

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DependencyFormatValidator.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Created by Genadi Rabinovich, genadi@hpe.com on 23/06/2016.
+ */
+class DependencyFormatValidator {
+    private static final int DEPENDENCY_PARTS = 3;
+    public static final String INVALID_DEPENDENCY = "Dependency definition should contain exactly [" + DEPENDENCY_PARTS + "] non empty parts separated by ':'";
+    void validateDependency(String dependency) {
```
Comment (Bonczidai) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DependencyFormatValidator.java]: maybe add this method to the `PreCompileValidator` so all the validations will be in one central place - WDYT?

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformer.java [hunk 2]
```diff
@@ -17,55 +17,45 @@
...
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import org.springframework.stereotype.Component;
 
 @Component
 public class JavaActionTransformer extends AbstractTransformer implements Transformer<Map<String, String>, Map<String, String>> {
-
+    private static DependencyFormatValidator dependencyFormatValidator = new DependencyFormatValidator();
```
Comment (Bonczidai) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformer.java]: I think we could define this class as a spring bean (@Component), the autowire it (if you don't go with the `PreCompileValidator` approach)

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/PythonActionTransformer.java [hunk 3]
```diff
@@ -17,15 +17,13 @@
...
-import org.springframework.stereotype.Component;
+import java.util.*;
 
-import java.util.Map;
-import java.util.List;
+import org.springframework.stereotype.Component;
 
 @Component
 public class PythonActionTransformer extends AbstractTransformer implements Transformer<Map<String, Serializable>, Map<String, Serializable>> {
+    private static DependencyFormatValidator dependencyFormatValidator = new DependencyFormatValidator();
```
Comment (Bonczidai) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/PythonActionTransformer.java]: I think we could define this class as a spring bean (@Component), the autowire it (if you don't go with the `PreCompileValidator` approach)

File: cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformerTest.java [hunk 4]
```diff
@@ -39,6 +39,7 @@
  *
  * @author Bonczidai Levente
  */
+@SuppressWarnings("ALL")
```
Comment (Bonczidai) [inline cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformerTest.java]: suppress 'ALL' - is it a good idea?

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DependencyFormatValidator.java (hunk 1 above)
Comment (genadi-hp) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DependencyFormatValidator.java]: Defined as bean

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformer.java (hunk 2 above)
Comment (genadi-hp) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformer.java]: Defined as bean and autowared

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/PythonActionTransformer.java (hunk 3 above)
Comment (genadi-hp) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/PythonActionTransformer.java]: Defined as bean and autowared

File: cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformerTest.java (hunk 4 above)
Comment (genadi-hp) [inline cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/modeller/transformers/JavaActionTransformerTest.java]: You are right, done by mistake.
Handling it in better way

**Comment by Bonczidai** [comment]
:+1:

**Pull request merged**